library-udf/src/test/java/org/apache/iotdb/library/dmatch/DMatchTests.java

```diff
@@ -0,0 +1,782 @@
+              500, 1, 1, 1, 1));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d3(timestamp,s1,s2,s3,s4) values(%d,%d,%d,%d,%d)",
+              600, 1, 1, 1, 1));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d3(timestamp,s1,s2,s3,s4) values(%d,%d,%d,%d,%d)",
+              700, 1, 1, 1, 1));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d3(timestamp,s1,s2,s3,s4) values(%d,%d,%d,%d,%d)",
+              800, 2, 2, 2, 2));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d3(timestamp,s1,s2,s3,s4) values(%d,%d,%d,%d,%d)",
+              900, 3, 3, 3, 3));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d3(timestamp,s1,s2,s3,s4) values(%d,%d,%d,%d,%d)",
+              1000, 2, 2, 2, 2));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d3(timestamp,s1,s2,s3,s4) values(%d,%d,%d,%d,%d)",
+              1100, 1, 1, 1, 1));
+
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              100, 0, 0, 0, 0, 6, 6, 6, 6));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              200, 2, 2, 2, 2, 7, 7, 7, 7));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              300, 3, 3, 3, 3, 0, 0, 0, 0));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              400, 4, 4, 4, 4, 9, 9, 9, 9));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d4(timestamp,s1,s2,s3,s4,s5,s6,s7,s8) values(%d,%d,%d,%d,%d,%d,%d,%d,%d))",
+              500, 5, 5, 5, 5, 10, 10, 10, 10));
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void registerUDF() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "create function completeness as 'org.apache.iotdb.library.dmatch.UDAFCov'");
+      statement.execute("create function timeliness as 'org.apache.iotdb.library.dmatch.UDAFDtw'");
+      statement.execute(
+          "create function consistency as 'org.apache.iotdb.library.dmatch.UDAFPearson'");
+      statement.execute("create function validity as 'org.apache.iotdb.library.dmatch.UDTFPtnSym'");
+      statement.execute("create function validity as 'org.apache.iotdb.library.dmatch.UDTFXcorr'");
```

Comment:
Validity has already been registered.Is this a mistake?

Reply:
Done.

Reply:
This has not been changed yet.

Reply:
Sorry it's my mistake

Reply:
Thank you！